dbms/src/Core/Block.cpp

```diff
@@ -543,8 +543,9 @@ Block mergeBlocks(Blocks && blocks)
 
     for (size_t i = 1; i < blocks.size(); ++i)
     {
-        if (likely(blocks[i].rows()) > 0)
+        if (likely(blocks[i].rows() > 0))
         {
+            RUNTIME_CHECK_MSG(blocksHaveEqualStructure(first_block, blocks[i]), "Cannot merge different structure blocks.");
```

Comment:
Why we need to check structure? What will happen when two blocks' structure are different?

Reply:
> Why we need to check structure? What will happen when two blocks' structure are different?

This is for protection, anyone can call mergeBlocks, if there is a different structure in the input blocks, the result will be problematic.  Its function is  similar here https://github.com/pingcap/tiflash/blob/fececd9375bd3b5de2318788bb08fa84e7c5deed/dbms/src/Interpreters/Aggregator.cpp#L2042

Reply:
> Why we need to check structure? What will happen when two blocks' structure are different?

if the column of block[0] is nullable, block[1] is not nullable.  the merging will transfer the column of block[1] to nullable and insert to the column block[0]. You can't be sure if the column we need will be nullable or not.